test/Algebra/Graph/Test/Bipartite/AdjacencyMap.hs

```diff
@@ -0,0 +1,620 @@
+            == Map.fromAscList [(2, Set.fromAscList [1, 3]), (4, Set.fromAscList [1, 3])]
+    test "leftAdjacencyMap (fromGraph (biclique (map Left [1..x]) (map Right [1..y])))                                                         == <correct result>" $ \x y ->
+        (leftAdjacencyMap $ fromGraph $ G.biclique (map Left [1..(x :: Int)]) (map Right [1..(y :: Int)]))
+            == expectedBicliqueMap x y
+    test "rightAdjacencyMap (fromGraph (biclique (map Left [1..x]) (map Right [1..y])))                                                        == <correct result>" $ \x y ->
+        (rightAdjacencyMap $ fromGraph $ G.biclique (map Left [1..(x :: Int)]) (map Right [1..(y :: Int)]))
+            == expectedBicliqueMap y x
+
+    putStrLn "\n============ Bipartite.AdjacencyMap.fromBipartite ============"
+    test "fromBipartite empty                                                            == AM.empty" $
+        fromBipartite (empty :: BAII) == AM.empty
+    test "fromBipartite (leftVertex 1)                                                   == AM.vertex (Left 1)" $
+        fromBipartite (leftVertex 1 :: BAII) == AM.vertex (Left 1)
+    test "fromBipartite (rightVertex 1)                                                  == AM.vertex (Right 1)" $
+        fromBipartite (rightVertex 1 :: BAII) == (AM.vertex (Right 1))
+    test "fromBipartite (edge x y)                                                       == AM.edges [(Left x, Right y), (Right y, Left x)]" $ \x y ->
+        fromBipartite (edge x y :: BAII) == AM.edges [(Left x, Right y), (Right y, Left x)]
+    test "fromBipartite (toBipartite (AM.edges [(Left x, Right y), (Right y, Left x)]))  == AM.edges [(Left x, Right y), (Right y, Left x)]" $ \x y ->
+        (fromBipartite $ toBipartite (AM.edges [(Left x, Right y), (Right y, Left x)] :: AII)) == AM.edges [(Left x, Right y), (Right y, Left x)]
+    test "fromBipartite (edges [(1, 1), (1, 2)])  == <correct result>" $
+        fromBipartite (edges [(1, 1), (1, 2)] :: BAII)
+            == (AM.edges [(Left 1, Right 1), (Left 1, Right 2), (Right 1, Left 1), (Right 2, Left 1)])
+    test "AM.consistent (fromBipartite x)                                                == True" $ \x ->
+        AM.consistent $ fromBipartite (x :: BAII)
+    test "fromBipartite (toBipartite (AM.biclique (map Left [1..x]) (map Right [1..y]))) == <correct result>" $ \x y ->
+        (fromBipartite $ toBipartite $ AM.biclique (map Left [1..(x :: Int)]) (map Right [1..(y :: Int)]))
+         == AM.overlay (AM.biclique (map Left [1..(x :: Int)]) (map Right [1..(y :: Int)]))
+                       (AM.biclique (map Right [1..(y :: Int)]) (map Left [1..(x :: Int)]))
+
+    putStrLn "\n============ Bipartite.AdjacencyMap.hasEdge ============"
+    test "hasEdge x y (fromGraph empty)                                 == False" $ \x y ->
+        not $ hasEdge (x :: Int) (y :: Int) $ fromGraph G.empty
```

Comment:
It's typically a bit easier to read if these annotations are on the lambda, i.e. `\(x :: Int) (y :: Int) ->...`.